Mathlib/Analysis/SpecialFunctions/ImproperIntegrals.lean

```diff
@@ -177,3 +178,30 @@ theorem integral_Ioi_cpow_of_lt {a : ℂ} (ha : a.re < -1) {c : ℝ} (hc : 0 < c
   simp_rw [neg_neg, Complex.norm_eq_abs, Complex.abs_cpow_eq_rpow_re_of_pos hx, Complex.add_re,
     Complex.one_re]
 #align integral_Ioi_cpow_of_lt integral_Ioi_cpow_of_lt
+
+theorem intervalIntegral_inv_one_add_sq_tendsto :
+    Tendsto (fun i => ∫ (x : ℝ) in -i..i, (1 + x ^ 2)⁻¹) atTop (𝓝[<] π) := by
+  convert TendstoNhdsWithinIio.const_mul (b := 2) (by norm_num) arctan_atTop <;> simp [two_mul]
+
+theorem integrable_inv_one_add_sq : Integrable fun (x : ℝ) ↦ (1 + x ^ 2)⁻¹ := by
+  suffices Integrable fun (x : ℝ) ↦ (1 + ‖x‖ ^ 2) ^ ((-2 : ℝ) / 2) by simpa [rpow_neg_one]
+  exact integrable_rpow_neg_one_add_norm_sq (by simpa using by norm_num)
+
+@[simp]
+theorem integral_Iic_inv_one_add_sq {i : ℝ} :
+    ∫ (x : ℝ) in Set.Iic i, (1 + x ^ 2)⁻¹ = arctan i + (π / 2) :=
+  integral_Iic_of_hasDerivAt_of_tendsto' (fun _ _ => by rw [← one_div]; apply hasDerivAt_arctan)
+    integrable_inv_one_add_sq.integrableOn (tendsto_nhds_of_tendsto_nhdsWithin arctan_atBot)
+    |>.trans (sub_neg_eq_add _ _)
+
+@[simp]
+theorem integral_Ioi_inv_one_add_sq {i : ℝ} :
+    ∫ (x : ℝ) in Set.Ioi i, (1 + x ^ 2)⁻¹ = (π / 2) - arctan i :=
+  integral_Ioi_of_hasDerivAt_of_tendsto' (fun _ _ => by rw [← one_div]; apply hasDerivAt_arctan)
+    integrable_inv_one_add_sq.integrableOn (tendsto_nhds_of_tendsto_nhdsWithin arctan_atTop)
+
+@[simp]
+theorem integral_volume_inv_one_add_sq : ∫ (x : ℝ), (1 + x ^ 2)⁻¹ = π :=
+  tendsto_nhds_unique
```

Comment:
It looks like we are missing a lemma `integral_of_hasDerivAt_of_tendsto'` similar to the ones on`Ioi` and `Iic` but for the whole line. Do you think you could add it, and use it to derive the result here (probably dropping `intervalIntegral_inv_one_add_sq_tendsto`)?